src/cirrus/plugins/publishers/documentation.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python
+"""
+_documentation_
+
+Publisher plugin that uses fabric to do a remote put
+
+"""
+from fabric.operations import put
+from cirrus.fabric_helpers import FabricHelper
+from cirrus.logger import get_logger
+from cirrus.publish_plugins import Publisher
+
+
+LOGGER = get_logger()
+
+
+class Documentation(Publisher):
+    PLUGGAGE_OBJECT_NAME = 'file_server'
+
+    def publish(self, opts, doc_artifact):
+        """
+        publish docs to a remote server via fabric over ssh.
+
+        .. note:: This only uploads the doc artifact to the server. If
+            desired, the artifact must be manually unpacked on the server.
+
+        Requires the following cirrus.conf sections:
+        [doc]
+        publisher = file_server
+
+        [file_server]
+        file_server_url = http://localhost:8080
+        file_server_upload_path = /path/to/upload/dir
+        file_server_sudo = [True | False]
+
+        .. note:: file_server_sudo is optional for False because it defaults
+            to False.
+
+        Requires the following .gitconfig options in the [cirrus] section:
+        file-server-username = username
+        file-server-keyfile = /path/to/keyfile
+
+        """
+        fs_creds = self.package_conf.credentials.file_server_credentials()
+        fs_username = fs_creds['file_server_username']
+        fs_keyfile = file_server_creds['file_server_keyfile']
+
+        try:
+            fs_config = self.package_conf['file_server']
+            fs_url = fs_config['file_server_url']
+            fs_upload_path = fs_config['file_server_upload_path']
+            fs_sudo = fs_config.get('file_server_sudo', False)
+        except KeyError:
```

Comment:
if you trap this as ex, you can tell the user which key was missing in the error message